components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -4104,6 +4104,36 @@ public static NativeArray jsFunction_getTiers(Context cx, Scriptable thisObj, Ob
         return myn;
     }
 
+    public static NativeArray jsFunction_getAppTiers(Context cx, Scriptable thisObj, Object[] args, Function funObj) {
+        NativeArray myn = new NativeArray(0);
+        APIConsumer apiConsumer = getAPIConsumer(thisObj);
+        Set<Tier> tiers;
+        try {
+            //If tenant domain is present in url we will use it to get available tiers
+            if (args.length > 0 && args[0] != null) {
+                tiers = apiConsumer.getAppTiers((String) args[0]);
+            } else {
+                tiers = apiConsumer.getAppTiers();
+            }
+
+            List<Tier> tierList = APIUtil.sortTiers(tiers);
+            int i = 0;
+            for (Tier tier : tierList) {
+                NativeObject row = new NativeObject();
+                row.put("tierName", row, tier.getName());
+                row.put("tierDisplayName", row, tier.getDisplayName());
+                row.put("tierDescription", row, tier.getDescription() != null ? tier.getDescription() : "");
+                row.put("defaultTier", row, i == 0);
+                myn.put(i, myn, row);
+                i++;
+            }
+
+        } catch (Exception e) {
```

Comment:
Same as above. Mention why we're catching generic Exception and the reason for continuing the process.